### PR TITLE
Fix for issue regarding function evaluation

### PR DESCRIPTION
**Pull request by msaxena2**
Proposed Fix for #1902 . WIP, not yet ready for review.

### CODE REVIEW

**Comment by msaxena2** [comment]
@cos Please review.

**Comment by msaxena2** [comment]
@cos Nevermind, looks like this still needs some more work.

**Comment by msaxena2** [comment]
@cos I've made some more changes that I think work. Please review.

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/CopyOnWriteTransformer.java [hunk 1]
```diff
@@ -179,12 +207,56 @@ public ASTNode transform(KItemProjection kItemProjection) {
...
+    private Term normalizeKSeqList(KList kList) {
+        if (kList.size() > 1 && (kList.get(0) instanceof KItem) && ((KItem) (kList.get(0))).klabel().name().equals(KSEQUENCE_KLABEL)) {
+            KItem kSeq = (KItem) (kList.get(0));
+            if (kSeq.kList() instanceof KList) {
+                KList kSeqList = (KList) kSeq.klist();
+                Term rightNormalizedChild = addRightAssoc(kSeqList.get(1), kList.get(1));
+                KList.Builder builder = KList.builder();
+                builder.concatenate(kSeqList.get(0));
+                builder.concatenate(rightNormalizedChild);
+                return builder.build();
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/CopyOnWriteTransformer.java:230]: the above four lines can be done with something like `KList.concatenate`

File: kernel/src/main/java/org/kframework/utils/OneIntegerGenericBitSet.java [hunk 2]
```diff
@@ -239,7 +239,17 @@ else if (count == 1) {
...
+            return true;
+        if(count == 1)
+            return bitSet.get(oneValue);
+        if(bitSet.count <= 1)
+            return false;
+
+        java.util.BitSet x = new java.util.BitSet();
+        x.or(this.severalValues);
+        x.and(bitSet.severalValues);
+        return x.equals(this.severalValues);
```
Comment (cos) [inline kernel/src/main/java/org/kframework/utils/OneIntegerGenericBitSet.java:252]: Could you move this in it's own PR? It has nothing to do with function evaluation. It looks like the code I wrote to make larger bitsets work.

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/CopyOnWriteTransformer.java [hunk 3]
```diff
@@ -179,12 +207,56 @@ public ASTNode transform(KItemProjection kItemProjection) {
...
+
+    private Term addRightAssoc(Term term, Term toBeAdded) {
+        if (term instanceof KItem && ((KItem) term).klabel().name().equals(KSEQUENCE_KLABEL)) {
+            KItem kItem = (KItem) term;
+            if (kItem.klist() instanceof KList) {
+                KList kList = (KList) kItem.kList();
+                Term rightTerm = addRightAssoc(kList.get(1), toBeAdded);
+                KList.Builder builder = KList.builder();
+                builder.concatenate(kList.get(0));
+                builder.concatenate(rightTerm);
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/CopyOnWriteTransformer.java:244]: `KList.concatenate` here also

File: java-backend/src/main/java/org/kframework/backend/java/symbolic/CopyOnWriteTransformer.java [hunk 4]
```diff
@@ -179,12 +207,56 @@ public ASTNode transform(KItemProjection kItemProjection) {
...
+                return KItem.of((Term) kItem.klabel(), builder.build(), kItem.globalContext(),
+                        kItem.getSource(), kItem.location());
+            }
+            return kItem;
+        }
+        //construct new KSequence Term
+        KList.Builder builder = KList.builder();
+        builder.concatenate(term);
+        builder.concatenate(toBeAdded);
+        GlobalContext globalContext = term instanceof HasGlobalContext ?
```
Comment (cos) [inline java-backend/src/main/java/org/kframework/backend/java/symbolic/CopyOnWriteTransformer.java:254]: `KList.concatenate` here also

**Comment by cos** [comment]
Looks good overall. One more thing: could you make sure that this doesn't impact performance significantly? Try to run any definition before and after on a long-running program (e.g., counting to 10k).
